sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -311,6 +311,51 @@ namespace Core { namespace _internal {
      */
     ~PosixTimeConverter() = delete;
   };
+
+  /**
+   * @brief Provides convertion methods for Windows NT filetime to an #Azure::DateTime.
+   *
+   */
+  class NTFileTimeTimeConverter final {
```

Comment:
`TimeTime`?

I think we should workshop the name a bit. Maybe we can avoid the window-ism, if this format is more general purpose.

Reply:
See @JeffreyRichter's email. The NT typename is "FILETIME", so this converter is a TimeConverter which converts NT FILETIMES.

Would you rather Win32FILETIME instead of NTFILETIME?

Reply:
The documentation frequently uses the term `file time` on its own. And some search term hits show people refer to it as Windows filetime. Probably Win32/NT is implied by the context:
https://learn.microsoft.com/en-us/windows/win32/sysinfo/file-times
https://stackoverflow.com/questions/6161776/convert-windows-filetime-to-second-in-unix-linux

That said though, just `FileTimeConverter` would be too generic and take-up a good name.

> Would you rather Win32FILETIME instead of NTFILETIME?

Yeah, between Win32 vs NT, I would lean more towards Win32 to mirror Posix rather than NT. That seems more approachable and self-descriptive to me, but you are way more familiar with this space, so feel free to make the call here.

**Options:**
- `Win32FileTimeConverter` - my preference
- `NTFileTimeConverter`

Reply:
I'm actually happier with WIn32 than I am with NT - Although it is a purely historical artifact, Win32 is actually similar to Posix - it's a description of an API surface which can be implemented on multiple platforms (originally there was something called Win32s which allowed Win32 applications to run on Win16 for example). And Win95 ran Win32 apps just fine.

Reply:
@JeffreyRichter Can you make the call for the class name?